src/maindx.cpp

```diff
@@ -1063,49 +1016,19 @@ extern "C" void ADDON_Destroy()
   initialized = false;
 }
 
-//-- GetStatus ---------------------------------------------------------------
-// Returns the current Status of this visualisation
-// !!! Add-on master function !!!
-//-----------------------------------------------------------------------------
-extern "C" ADDON_STATUS ADDON_GetStatus()
-{
-  return ADDON_STATUS_OK;
-}
-
 //-- SetSetting ---------------------------------------------------------------
 // Set a specific Setting value (called from XBMC)
 // !!! Add-on master function !!!
 //-----------------------------------------------------------------------------
-extern "C" ADDON_STATUS ADDON_SetSetting(const char *strSetting, const void* value)
+ADDON_STATUS CVisualizationShadertoy::SetSetting(const std::string& settingName, const kodi::CSettingValue& settingValue)
```

Comment:
Yes, is in settings.xml only declared as `<setting id="lastpresetidx" type="number" option="int" default="1" visible="false" />` with visible as false and can never changed by the user where SetSetting is needed.